xdrip/Managers/Alerts/AlertManager.swift

```diff
@@ -137,27 +137,16 @@ public class AlertManager:NSObject {
                 if latestBgReadings.count > 1 {
                     lastButOneBgREading = latestBgReadings[1]
                 }
-                
-                
-                // alerts are checked in order of importance - there should be only one alert raised, except missed reading alert which will always be checked.
-                // first check very low alert
-                if (!checkAlertAndFire(alertKind: .verylow, lastBgReading: lastBgReading, lastButOneBgREading: lastButOneBgREading, lastCalibration: lastCalibration, transmitterBatteryInfo: transmitterBatteryInfo)) {
-                    // very low not fired, check low alert - if very low alert snoozed, skip the check for low alert and continue to next step
-                    if getSnoozeParameters(alertKind: AlertKind.verylow).getSnoozeValue().isSnoozed || (!checkAlertAndFire(alertKind: .low, lastBgReading: lastBgReading, lastButOneBgREading: lastButOneBgREading, lastCalibration: lastCalibration, transmitterBatteryInfo: transmitterBatteryInfo)) {
-                        //  low not fired, check very high alert
-                        if (!checkAlertAndFire(alertKind: .veryhigh, lastBgReading: lastBgReading, lastButOneBgREading: lastButOneBgREading, lastCalibration: lastCalibration, transmitterBatteryInfo: transmitterBatteryInfo)) {
-                            // very high not fired, check high alert - if very high alert snoozed, skip the check for high alert and continue to next step
-                            if getSnoozeParameters(alertKind: AlertKind.veryhigh).getSnoozeValue().isSnoozed || (!checkAlertAndFire(alertKind: .high, lastBgReading: lastBgReading, lastButOneBgREading: lastButOneBgREading, lastCalibration: lastCalibration, transmitterBatteryInfo: transmitterBatteryInfo)) {
-                                // very high not fired check calibration alert
-                                if (!checkAlertAndFire(alertKind: .calibration, lastBgReading: lastBgReading, lastButOneBgREading: lastButOneBgREading, lastCalibration: lastCalibration, transmitterBatteryInfo: transmitterBatteryInfo)) {
-                                    // finally let's check the battery level alert
-                                    _ = checkAlertAndFire(alertKind: .batterylow, lastBgReading: lastBgReading, lastButOneBgREading: lastButOneBgREading, lastCalibration: lastCalibration, transmitterBatteryInfo: transmitterBatteryInfo)
-                                }
-                            }
-                        }
-                    }
-                }
-                // set missed reading alert, this will be a future planned alert
+
+                // TODO original implementation ignores isSnoozed for .calibration and .batterylow
```

Comment:
to snooze status is checked in function checkAlertAndFire, also for battery and calibration
The additional check for isSnoozed is not there to check if the alert itself is snoozed but if a higher level alert is snoozed.

Example, very low is snoozed, will proceed to next alert with is low. But if user has snoozed verylow, then probably he or she doesn't want to have an alert for low

Reply:
Ah that makes, I've updated the code accordingly